lib/kadm5/rename_s.c

```diff
@@ -116,7 +116,9 @@ kadm5_s_rename_principal(void *server_handle,
 
     /* NOTE: We do not use hdb_fetch_kvno() here */
     ret = context->db->hdb_fetch_kvno(context->context, context->db,
-				      source, HDB_F_GET_ANY|HDB_F_ADMIN_DATA, 0, &ent);
+                                      source,
+                                      HDB_F_DECRYPT|HDB_F_GET_ANY|HDB_F_ADMIN_DATA,
```

Comment:
W.r.t. all the missing `HDB_F_DECRYPT`s, maybe it was working by accident for users who encrypt keys in the HDB, and maybe adding them here makes it not encrypt keys by accident (some such problem) when we want then encrypted.